### DISCUSSION TITLE
error compiling for device

### DISCUSSION THREAD

**Discussion by feclare** [boozook/playdate#421] [Q&A] [answered]
Hi, I'm having problems building for the device. Building for the simulator works fine. 
```bash
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.42s
   Compiling staticlib of playdate v0.1.0 (/home/user/projects/sweetbomb-rs/playdate)
/usr/lib/gcc/arm-none-eabi/14.1.0/../../../../arm-none-eabi/bin/ld: cannot find -lc: No such file or directory
collect2: error: ld returned 1 exit status
error: process exited unsuccessfully: exit status: 1
~
```
I'm using arch linux and I think I installed all the required packages.
Any idea of what I'm doing wrong?

**Comment by andreascasapu** [comment]
If I am understanding correctly what the error means, I think it's not able to find the static libc library for that architecture/gcc version.

I think I ran into something similar when compiling for the emulator. My problem, if I recall correctly, was that I hadn't fully installed all the dev libraries for the right version of gcc.

You might want to try looking at your installed packages for versions of gcc/g++/build-essential/etc. and see if you are missing something for arm-none-eabi.

I am not entirely sure that this is the problem/solution, but I hope that it leads you in the right direction.

**Reply by boozook** [reply]
I also supposed that it’s so.
Also you can try to build bin cargo-target (`crate-type = bin)` without gcc, with just rust+llvm with `--no-gcc` arg.

**Reply by feclare** [reply, marked as answer]
I was missing the libc package for `arm-none-eabi`. Instaling `arm-none-eabi-newlib` and `arm-none-eabi-gcc` both fixed my linking problem in arch. 

Thanks
